app/src/main/java/org/jellyfin/androidtv/details/TrailerPresenter.kt

```diff
@@ -0,0 +1,149 @@
+package org.jellyfin.androidtv.details
+
+import android.content.Context
+import android.content.Intent
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import android.widget.ImageView
+import android.widget.Toast
+import androidx.appcompat.view.ContextThemeWrapper
+import androidx.leanback.widget.Presenter
+import com.squareup.picasso.Picasso
+import kotlinx.android.synthetic.main.multi_badge_image_card_view.view.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.TvApp
+import org.jellyfin.androidtv.base.IItemClickListener
+import org.jellyfin.androidtv.model.itemtypes.BaseItem
+import org.jellyfin.androidtv.model.itemtypes.PlayableItem
+import org.jellyfin.androidtv.ui.FavoriteBadge
+import org.jellyfin.androidtv.ui.MultiBadgeImageCardView
+import org.jellyfin.androidtv.ui.WatchedBadge
+import org.jellyfin.androidtv.util.ImageUtils
+import org.jellyfin.androidtv.util.dp
+import org.jellyfin.apiclient.model.entities.MediaUrl
+
+private const val SAFE_BRANDING_COMPLIANCE: Boolean = true
+
+class TrailerPresenter(private val context: Context, private val imageHeight: Int) : Presenter(), IItemClickListener {
```

Comment:
At this moment the server only provides youtube trailers but a new plugin could easily add urls for Vevo or other sites/apps. I think we should prepare for this by abstracting the external media.

A separate class could be made for `YoutubeTrailer` that extends a class `ExternalTrailer`. Some magic could convert the info we have to the youtube trailer class and that one contains some functions like `.open()` `.getThumbnail()` `.getBrandingIcon()`. Or something similar.

Next, we do some calculations to get the video-id from youtube urls. I want unit tests for this so it's properly tested.